File: src/main/java/com/hy/http/RequestMethod.java
package com.hy.http;

/**
 * RequestMethod HTTP request method.
 *
 * @author HeYan
 * @time 2017/5/24 11:41
 */
public enum RequestMethod {

    GET("GET"),

    POST("POST"),

    PUT("PUT"),

    DELETE("DELETE"),

    HEAD("HEAD"),

    OPTIONS("OPTIONS"),

    TRACE("TRACE"),

    PATCH("PATCH");

    private final String value;

    RequestMethod(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return this.value;
    }

}

File: src/main/java/com/hy/http/RequestMethod.kt
package com.hy.http

/**
 * RequestMethod HTTP request method.

 * @author HeYan
 * *
 * @time 2017/5/24 11:41
 */
enum class RequestMethod constructor(private val value: String) {

    GET("GET"),

    POST("POST"),

    PUT("PUT"),

    DELETE("DELETE"),

    HEAD("HEAD"),

    OPTIONS("OPTIONS"),

    TRACE("TRACE"),

    PATCH("PATCH");

    override fun toString(): String {
        return this.value
    }

}
